streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -655,25 +673,27 @@ public Subscription subscription(final Set<String> topics) {
                 }
             }
         }
+
         taskManager.setPartitionsByHostState(partitionsByHostState);
 
         final Map<String, Assignment> assignment;
         if (versionProbing) {
-            assignment = versionProbingAssignment(clientMetadataMap, partitionsForTask, partitionsByHostState, futureConsumers, minReceivedMetadataVersion);
+            assignment = versionProbingAssignment(clientMetadataMap, partitionsForTask, tasksByHostState, futureConsumers, minReceivedMetadataVersion);
```

Comment:
In the case of version probing, we send an old assignment back. If we encode the new `tasksByHostState` instead of `partitionsByHostState` wouldn't the instances that are not upgrade yet crash?

Or course, to be future prove, eg, we bump the version from 5 to 6, it might be ok for a 5 to 6 upgrade to send the new encoding. We need to make the cut-off point base on the smallest version we received.

Reply:
I have added partitionsByHostState in the assignmentInfo to make it backward compatible. And reverted how older versions are encoded and decoded.